File: spark/src/main/java/com/jason/spark/core/Spark06RDDGroupBy.java
package com.jason.spark.core;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;

import java.util.Arrays;

/**
 * @Description:
 * @author: 贾森
 * @date: 2024年10月25日 17:19
 */
public class Spark06RDDGroupBy {
    public static void main(String[] args) {
        SparkConf sparkConf = new SparkConf().setMaster("local[*]").setAppName("Spark01 RDD Operator Transform");
        JavaSparkContext sc = new JavaSparkContext(sparkConf);
        JavaRDD<Integer> parallelize = sc.parallelize(Arrays.asList(1, 2, 3, 4, 5, 6), 2);
        // 泛型为分组标记的类型
        JavaPairRDD<Integer, Iterable<Integer>> groupByRDD = parallelize.groupBy( integer -> integer % 2);
        groupByRDD.collect().forEach(System.out::println);

        // 类型可以任意修改
        JavaPairRDD<Boolean, Iterable<Integer>> groupByRDD1 = parallelize.groupBy(value -> value % 2 == 0);
        groupByRDD1.collect().forEach(System.out::println);
        sc.stop();
    }
}

File: spark/src/main/scala/com/jason/spark/core/rdd/Spark06_RDD_GroupBy.scala
package com.jason.spark.core

import org.apache.spark.{SparkConf, SparkContext}

object Spark06_RDD_GroupBy {
  def main(args: Array[String]): Unit = {
    val sparkConf = new SparkConf().setMaster("local[*]").setAppName("Spark06_RDD_GroupBy")
    val sc = new SparkContext(sparkConf)
    val rdd = sc.makeRDD(List(1, 2, 3, 4, 5, 6),2)
    val groupbyRDD = rdd.groupBy(v => v % 2)
    groupbyRDD.collect().foreach(println)

    val groupbyRDD1 = rdd.groupBy(v => v % 2 == 0)
    groupbyRDD1.collect().foreach(println)

  }

}
